srml/system/src/lib.rs

```diff
@@ -1184,7 +1217,7 @@ mod tests {
 		fn from(e: Event) -> u16 {
 			match e {
 				Event::ExtrinsicSuccess => 100,
-				Event::ExtrinsicFailed => 101,
+				Event::ExtrinsicFailed(err) => Encode::using_encoded(&err, |s| (s[0] as u16) | ((s[1] as u16) << 8)),
```

Comment:
What's the reasoning behind this encoding?

Reply:
That is some leftover code from the initial version of @xlc. Removed it.

Reply:
this is just a way to convert `DispatchError` to `u16`.  DispatchError should serialize to a two byte value (according to the initial discussion, which I am not sure is still the case now)

Reply:
It is now 3 bytes, but the manual conversion isn't required anymore.